examples/vision/bit.py

```diff
@@ -0,0 +1,308 @@
+## Counting number of images per class
+"""
+
+labels = []
+for image_path in image_paths:
+    label = image_path.split("/")[1]
+    labels.append(label)
+class_count = Counter(labels)
+pprint(class_count)
+
+"""
+Imbalance Dataset
+"""
+
+"""
+## Define hyperparameters
+"""
+
+RESIZE_TO = 260
+CROP_TO = 224
+TRAIN_SPLIT = 0.9
+BATCH_SIZE = 128
+AUTO = tf.data.AUTOTUNE
+NUM_CLASSES = 30
+STEPS_PER_EPOCH = 140
+
+"""
+## Splitting the Dataset
+"""
+
+i = int(len(image_paths) * TRAIN_SPLIT)
+
+train_paths = image_paths[:i]
+train_labels = labels[:i]
+validation_paths = image_paths[i:]
+validation_labels = labels[i:]
+
+print(len(train_paths), len(validation_paths))
+
+"""
+## Encding the Labels
+"""
+
+le = LabelEncoder()
+train_labels_le = le.fit_transform(train_labels)
+validation_labels_le = le.transform(validation_labels)
+print(train_labels_le[:5])
+
+"""
+Since the dataset has class imbalance issue, it's good to supply class weights while training the model.
+"""
+
+trainLabels = to_categorical(train_labels_le)
+classTotals = trainLabels.sum(axis=0)
+classWeight = dict()
+# loop over all classes and calculate the class weight
+for i in range(0, len(classTotals)):
+    classWeight[i] = classTotals.max() / classTotals[i]
+
+"""
+## Set dataset-dependent hyperparameters
+"""
+
+DATASET_SIZE = "\u003C20k examples"  # @param ["<20k examples", "20k-500k examples", ">500k examples"]
+
+if DATASET_SIZE == "<20k examples":
+    SCHEDULE_LENGTH = 500
+    SCHEDULE_BOUNDARIES = [200, 300, 400]
+elif DATASET_SIZE == "20k-500k examples":
+    SCHEDULE_LENGTH = 10000
+    SCHEDULE_BOUNDARIES = [3000, 6000, 9000]
+else:
+    SCHEDULE_LENGTH = 20000
+    SCHEDULE_BOUNDARIES = [6000, 12000, 18000]
+
+"""
+## Convert the data into TensorFlow Dataset objects
+"""
+
+train_ds = tf.data.Dataset.from_tensor_slices((train_paths, train_labels_le))
+val_ds = tf.data.Dataset.from_tensor_slices((validation_paths, validation_labels_le))
+
+"""
+## Preprocessing helper functions
+"""
+
+SCHEDULE_LENGTH = SCHEDULE_LENGTH * 512 / BATCH_SIZE
+
+
+@tf.function
+def preprocess_train(image_path, label):
+    image = tf.io.read_file(image_path)
+    image = tf.image.decode_jpeg(image, channels=3)
+    image = tf.image.resize(image, (RESIZE_TO, RESIZE_TO))
+    image = tf.image.random_crop(image, [CROP_TO, CROP_TO, 3])
+    image = tf.cast(image, tf.float32) / 255.0
+    return (image, label)
+
+
+@tf.function
+def preprocess_test(image_path, label):
+    image = tf.io.read_file(image_path)
+    image = tf.image.decode_jpeg(image, channels=3)
+    image = tf.image.resize(image, (RESIZE_TO, RESIZE_TO))
+    image = tf.cast(image, tf.float32) / 255.0
+    return (image, label)
+
+
+DATASET_NUM_TRAIN_EXAMPLES = len(train_paths)
+
+"""
+## Create Data Pipeline for training
+"""
+
+# Training Pipeline
+pipeline_train = (
+    train_ds.shuffle(10000)
+    .repeat(
+        int(SCHEDULE_LENGTH * BATCH_SIZE / DATASET_NUM_TRAIN_EXAMPLES * STEPS_PER_EPOCH)
+        + 1
+        + 50
```

Comment:
Put this in a variable for better readability

Reply:
I am out of variable names can you kindly suggest me one to put there.

Reply:
```
repeat_count = int(
    SCHEDULE_LENGTH * BATCH_SIZE / DATASET_NUM_TRAIN_EXAMPLES * STEPS_PER_EPOCH)
repeat_count += 51
```

Also, why 51?